security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient_test.go

```diff
@@ -17,34 +17,28 @@ package stsclient
 import (
 	"context"
 	"testing"
-
-	"istio.io/istio/security/pkg/nodeagent/plugin/providers/google/stsclient/test"
 )
 
 func TestGetFederatedToken(t *testing.T) {
-	tlsFlag = false
-	defer func() {
-		tlsFlag = true
-	}()
-
 	r := NewPlugin()
 
-	ms, err := test.StartNewServer()
+	ms, err := StartNewServer(t)
+	if err != nil {
+		t.Fatalf("failed to start a mock server: %v", err)
+	}
 	secureTokenEndpoint = ms.URL + "/v1/identitybindingtoken"
 	defer func() {
-		ms.Stop()
+		if err := ms.Stop(); err != nil {
+			t.Logf("failed to stop mock server: %v", err)
+		}
 		secureTokenEndpoint = "https://securetoken.googleapis.com/v1/identitybindingtoken"
 	}()
 
-	if err != nil {
-		t.Fatalf("failed to start a mock server %v", err)
-	}
-
-	token, _, _, err := r.ExchangeToken(context.Background(), "", "")
+	token, _, _, err := r.ExchangeToken(context.Background(), fakeTrustDomain, fakeSubjectToken)
```

Comment:
trustDomain and subjectToken is not used at all? Not affecting the returned token I guess?

Reply:
I think it's used in for token exchange: https://github.com/istio/istio/blob/5b2f89555c4a8c354bc4878e4032b8810313f5b0/security/pkg/nodeagent/cache/secretcache.go#L708, the added tests will check it's set to the expected value in the exchange token request.